src/views/blocks/CarouselBlock.php

```diff
@@ -1,32 +1,70 @@
 <?php
 /* @var $this \luya\cms\base\PhpBlockView */
-
-if (!$this->isPrevEqual) {
-    $this->appView->registerJs("$('.carousel').carousel()");
+$config = '';
+if ($this->varValue('interval')) {
+    $config .= 'interval: '.$this->varValue('interval').',';
+}
+if ($this->varValue('keyboard')) {
+    $config .= 'keyboard: true,';
+}
+if ($this->varValue('pause')) {
+    $config .= 'pause: '.$this->varValue('pause').',';
+}
+if ($this->varValue('ride')) {
+    $config .= 'ride: '.$this->varValue('ride').',';
+}
+if ($this->varValue('wrap')==1) {
+    $config .= 'wrap: true,';
 }
+$config = rtrim($config, ',');
+$this->appView->registerJs("$('.carousel').carousel({".$config."})");
+$id = $this->extraValue('id');
+$images = $this->extraValue('images');
+$first = true;
+$indicators = '';
+$counter = 0;
 ?>
-<?php if (!$this->getIsPrevEqual()): ?>
-<div id="<?= $this->extraValue('id'); ?>" class="carousel slide" data-ride="carousel">
-	<div class="carousel-inner">
-<?php endif; ?>
-	<?php if ($this->extraValue('image')): ?>
-    <div class="carousel-item <?php if (!$this->getIsPrevEqual()): ?>active<?php endif; ?>">
-      	<img class="d-block w-100" src="<?= $this->extraValue('image')->source; ?>" alt="<?= $this->varValue('title'); ?>">
-      	<div class="carousel-caption d-none d-md-block">
-    	<h5><?= $this->varValue('title')?></h5>
-    	<p><?= $this->varValue('caption'); ?></p>
-  		</div>
-    </div>
+
+<div id="<?= $id ?>" class="carousel slide<?= $this->varValue('crossfade') ? ' carousel-fade' : '' ?><?= $this->varValue('row') ? ' row' : '' ?>" data-ride="carousel">
```

Comment:
You should use the the template parameter of varValue() for such things. For example 

```php
<?= $this->varValue('crossfade') ? ' carousel-fade' : '' ?>
```

could be

```
<?= $this->varValue('crossfade', null , 'carousel-fade'); ?>
```